api/proto/scheduler/scheduler.proto

```diff
@@ -0,0 +1,44 @@
+syntax = "proto3";
+
+enum RunnerType {
+  DOCKER = 0;
+}
+
+message Action {
+  string action_id = 1;
+  RunnerType runner_type = 2;
+  optional string image = 3;
```

Comment:
Assuming this is a container image, you should define a nested  `message ExecutionContext`:

```proto
 message Action {
  message ExecutionContext {
     RunnerType type = 1,
     optional string container_image = 2,
  }

  string action_id = 1;

```